lib/jsonrpc.ml

```diff
@@ -554,27 +554,41 @@ let get name dict =
     raise (Malformed_method_request str)
   | Some v -> v
 
-let call_of_string str =
+let version_and_call_of_string str =
   match of_string str with
   | Dict d ->
-    let name = match get "method" d with String s -> s | _ -> raise (Malformed_method_request str) in
-    let params =
+    let name =
+      match get "method" d with
+      | String s -> s
+      | _ -> raise (Malformed_method_request "method")
```

Comment:
I'd rather modify the message if the Malformed_method_request to "method" plus the str, because changing the type of the exception thrown affects the interface.